R/eatModel.r

```diff
@@ -1,55 +1,85 @@
-                return(list ( complete = dtmp[,c("score", "est", "bista", "ks")], short = shrt))}
+simEquiTable <- function(anchor, mRef, sdRef, addConst = 500, mulConst = 100, cutScores){
+  # checks/prep
+  anchor <- eatTools::makeDataFrame(anchor)
+  if(ncol(anchor) != 2){
+    warning(paste0("'anchor' has ", ncol(anchor)," columns. First column is used as item ID, second column is used as item parameter."))
+  }
+  if(!inherits(anchor[,2], c("integer", "numeric"))){
+    stop("Item parameter column must be numeric.")
+  }
+  if(length(unique(anchor[,1])) != nrow(anchor)){
+    stop("Item ID column has duplicated entries.")
+  }
+  lapply(c(mRef, sdRef, addConst, mulConst, cutScores), checkmate::assert_numeric, len = 1)
+
+  # function
+  dtmp <- data.frame(rbind(1*(lower.tri(matrix(1, nrow = nrow(anchor), ncol = nrow(anchor)))),1))
+  dtmp <- data.frame(dtmp, score = rowSums(dtmp), irtoys::wle(dtmp, cbind(1, anchor[,2], 0)), stringsAsFactors = FALSE)
+  dtmp[,"bista"] <- (dtmp[,"est"] - mRef) / sdRef * multConst + addConst
+  dtmp[,"ks"]    <- eatTools::num.to.cat(x = dtmp[,"bista"], cut.points = cutScores[["values"]], cat.values = cutScores[["labels"]])
+  dig <- 0
+
+  while (length(which(round(dtmp[,"bista"], digits = dig) %in% cutScores[["values"]])) > 0) {
+    dig <- dig + 1
+  }
+  shrt <- do.call("r.bind", by(data = dtmp, INDICES = dtmp[, "ks"],
+                               FUN = function(sks){
+                                 data.frame(score = paste(c(min(sks[, "score"]), max(sks[, "score"])), collapse = " bis "),
+                                            estimate = paste(round(c(min(sks[, "est"]), max(sks[, "est"])), digits = 2), collapse = " bis "),
+                                            bista = paste(round(c(min(sks[, "bista"]), max(sks[, "bista"])), digits = dig), collapse = " bis "),
+                                            ks = unique(sks[, "ks"]),
+                                            stringsAsFactors=FALSE)}))
+  return(list(complete = dtmp[,c("score", "est", "bista", "ks")], short = shrt))
+}
 
 ### not called -----------------------------------------------------------------
 
-getConquestVersion <- function ( path.conquest , path.temp , asDate = TRUE ) {
-    wd <- path.temp
-		f <- file.path ( wd , "delete.cqc" )
-		write ( "quit;" , f )
-		f <- normalizePath ( f )
-		path.conquest <- normalizePath ( path.conquest )
-		cmd <- paste ( "\"", path.conquest, "\" \"", f , "\"" , sep ="")
-		r <- NULL
-		suppressWarnings(try ( r <- system ( command = cmd , intern = TRUE ) , silent = TRUE ))
-		file.remove ( f )
-		if ( !is.null ( r ) ) {
-				r <- r[1]
-				r <- sub ( "ConQuest build: " , "" , r )
-				r <- gsub ( "\\s+" , "-" , r )
-				if ( asDate ) r <- date::as.date(r)
-		}
-		return (r)}
+getConquestVersion <- function(path.conquest, path.temp, asDate = TRUE){
+  # checks
+  lapply(c(path.conquest, path.temp), checkmate::assert_character, len = 1)

```

Comment:
is this right?

Reply:
That's not a user-level function. Hence, I would add assertations for the `conquest.folder` argument of the `defineModel()` function, but with `null.ok = TRUE`. Same is true for the `dir` argument of `defineModel()`. I don't know whether it's possible to test whether `dir` is a valid folder with writing permissions (which are necessary).

Reply:
I went through all the other functions and couldn't find any, that uses this function, though. 
defineModel() calls `identifyConquestFolder()` in line 12 (conquest.folder <- identifyConquestFolder()). 
But `identifyConquestFolder()` doesn't call `getConquestVersion()` either. 
it's the second function in [this R file](https://github.com/weirichs/eatModel/blob/master/R/defineModel_help.R)

is that function outdated or is there a typo somewhere? I always look through the code with strF, so maybe I overlooked something.

Reply:
Indeed, you are right. `getConquestVersion()` is outdated. You can delete the function and also delete the dependency `date` from the DESCRIPTION file. Thank you for pointing this out!